src/components/PaymentInfoLine.js

```diff
@@ -78,4 +76,170 @@ class PaymentInfoLine { // eslint-disable-line no-unused-vars
+                        }
+                        <div class="vendor-markup"></div>
+                    </template>
+                    <label class="highlight-on-bad-rate" data-i18n="payment-info-line-effective-rate">
+                        Effective rate
+                    </label>
+                    <div class="effective-rate highlight-on-bad-rate"></div>
+                </div>
+                <div class="rate-info info highlight-on-bad-rate"></div>
+                <div class="free-service-info info" data-i18n="payment-info-line-free-service">
+                    Nimiq provides this service free of charge.
+                </div>
+                <hr>
+                <div class="price-breakdown">
+                    <label data-i18n="payment-info-line-total">Total</label>
+                    <div class="total"></div>
+                </div>
+                <div class="network-fee-info info">
+                    + <span class="network-fee"></span>
+                    <span data-i18n="payment-info-line-network-fee">network fee</span>
+                </div>
+            </div>
+        `;
+        /* eslint-enable indent */
+
+        const formattedFiatAmount = NumberFormatting.formatCurrency(fiatAmount, fiatCurrency);
+        $tooltip.querySelectorAll('.fiat-amount').forEach($fiatAmount => {
+            $fiatAmount.textContent = formattedFiatAmount;
+        });
+
+        /** @type {HTMLTemplateElement} */
+        const $vendorMarkupTemplate = ($tooltip.querySelector('.vendor-markup-template'));
+        if (vendorMarkup !== undefined) {
+            // Convert to percent and round to two decimals. Always ceil to avoid displaying a lower fee than charged or
+            // larger discount than applied. Subtract small epsilon to avoid that numbers get rounded up as a result of
+            // floating point imprecision after multiplication. Otherwise formatting for example .07 results in 7.01%.
+            const vendorMarkupPercent = Math.ceil(vendorMarkup * 100 * 100 - 1e-10) / 100;
+            $vendorMarkupTemplate.replaceWith($vendorMarkupTemplate.content);
+            /** @type {HTMLElement} */
+            const $vendorMarkup = ($tooltip.querySelector('.vendor-markup'));
+            $vendorMarkup.textContent = `${vendorMarkup >= 0 ? '+' : ''}${vendorMarkupPercent}%`;
+        } else {
+            $vendorMarkupTemplate.remove();
+        }
+
+        /** @type {HTMLElement} */
+        const $effectiveRate = ($tooltip.querySelector('.effective-rate'));
+        // Fiat/crypto rate. Higher fiat/crypto rate means user is paying less crypto for the requested fiat amount
+        // and is therefore better for the user. Note: precision loss should be acceptable here.
+        const effectiveRate = fiatAmount / Nimiq.Policy.lunasToCoins(lunaAmount);
+        $effectiveRate.textContent = `${NumberFormatting.formatCurrency(effectiveRate, fiatCurrency, 0.0001)} / NIM`;
```

Comment:
Why a relative deviation of 0.0001? Why not simply 0?

Reply:
0 would mean printing up to 20 decimal positions which is an unnecessary high precision, especially if floating point precision errors occurred in the number to be displayed.